ee/clickhouse/models/property.py

```diff
@@ -225,50 +227,60 @@ def get_property_values_for_key(key: str, team: Team, value: Optional[str] = Non
     params = {}
     conditions = []
 
     if filters.get("selector"):
-        or_conditions = []
         selectors = filters["selector"] if isinstance(filters["selector"], list) else [filters["selector"]]
         for idx, query in enumerate(selectors):
             selector = Selector(query, escape_slashes=False)
             key = f"{prepend}_{idx}_selector_regex"
             params[key] = _create_regex(selector)
-            or_conditions.append(f"match(elements_chain, %({key})s)")
-        if len(or_conditions) > 0:
-            conditions.append("(" + (" OR ".join(or_conditions)) + ")")
+            conditions.append(f"match(elements_chain, %({key})s)")
+        if len(conditions) > 0:
+            conditions.append("(" + (" OR ".join(conditions)) + ")")
 
     if filters.get("tag_name"):
-        or_conditions = []
         tag_names = filters["tag_name"] if isinstance(filters["tag_name"], list) else [filters["tag_name"]]
         for idx, tag_name in enumerate(tag_names):
             key = f"{prepend}_{idx}_tag_name_regex"
             params[key] = rf"(^|;){tag_name}(\.|$|;|:)"
-            or_conditions.append(f"match(elements_chain, %({key})s)")
-        if len(or_conditions) > 0:
-            conditions.append("(" + (" OR ".join(or_conditions)) + ")")
+            conditions.append(f"match(elements_chain, %({key})s)")
+        if len(conditions) > 0:
+            conditions.append("(" + (" OR ".join(conditions)) + ")")
 
     attributes: Dict[str, List] = {}
-
     for key in ["href", "text"]:
-        vals = filters.get(key)
-        if filters.get(key):
-            attributes[key] = [re.escape(vals)] if isinstance(vals, str) else [re.escape(text) for text in filters[key]]
-
-    if len(attributes.keys()) > 0:
-        or_conditions = []
+        if vals := filters.get(key):
+            if operator.endswith("_set"):
+                attributes[key] = [r".*"]
+            else:
+                vals = [vals] if isinstance(vals, str) else vals
+                if operator.endswith("icontains"):
+                    attributes[key] = [f".*{re.escape(text)}.*" for text in vals]
+                elif operator.endswith("regex"):
+                    attributes[key] = vals
+                else:
+                    attributes[key] = [re.escape(text) for text in vals]
+    is_negated = operator in NEGATED_OPERATORS
+    if attributes:
         for key, value_list in attributes.items():
             for idx, value in enumerate(value_list):
-                params["{}_{}_{}_attributes_regex".format(prepend, key, idx)] = ".*?({}).*?".format(
-                    ".*?".join(['{}="{}"'.format(key, value)])
-                )
-                or_conditions.append("match(elements_chain, %({}_{}_{}_attributes_regex)s)".format(prepend, key, idx))
-            if len(or_conditions) > 0:
-                conditions.append("(" + (" OR ".join(or_conditions)) + ")")
+                optional_flag = "(?i)" if operator.endswith("icontains") else ""
```

Comment:
This is out of sync with what we were doing in property.py.

We should use `ILIKE` here as well instead of a custom solution.

Reply:
We already use regex for everything in here, I think this significantly simpler and less error-prone without an extra branch for `(not_)icontains`-only `ILIKE`.

Reply:
See the main comment I made around this - I think this is scope creep and shouldn't be in this PR.


That said, a proper solution would still unify this with `properties.py` in terms of approaches to icontains. Having similar, but different approaches is just a recipe for bugs/inconsistencies down the line.

Reply:
Normal property filtering involves values from parsed JSON though, while in this case we use the elements directly, so the benefit of a regex is that i.e. we can handle double quote characters better.